packages/types/executor/index.ts

```diff
@@ -0,0 +1,31 @@
+import { Step } from '../step';
+import { Task, TaskStepEventData } from '../task';
+import { Workflow } from '../workflow';
+
+interface Executor {
+  workflow: Workflow;
+}
+
+export interface TaskRepository {
+  save(task: Task): Promise<Task>;
+}
+
+class ExecutorImpl implements Executor {
+  public task: Task;
+  constructor(public workflow: Workflow, public taskRepository: TaskRepository) {
```

Comment:
@wtrocki do you think having a small repository interface a better solution?

This way executor can be back/frontend agnostic and then the implementation would be pushed up to the app, and pretty trivial.